lib/launchers/ocm.rb

```diff
@@ -0,0 +1,238 @@
+        @aws_access_key = ENV['AWS_ACCESS_KEY']
+        @aws_secret_key = ENV['AWS_SECRET_ACCESS_KEY'] || ENV['AWS_SECRET_KEY']
+      else
+        @cloud = ENV['OCM_CLOUD'] || @opts[:cloud]
+        if @cloud
+          @cloud_opts = default_opts(@cloud)
+          unless @cloud_opts
+            raise "Cannot find cloud '#{cloud}' defined in '#{service_name}'"
+          end
+          case @cloud_opts[:cloud_type]
+          when "aws"
+            aws = Amz_EC2.new(service_name: @cloud)            
+            @aws_account_id = aws.account_id
+            @aws_access_key = aws.access_key
+            @aws_secret_key = aws.secret_key
+          end
+        end
+      end
+    end
+
+    # @param service_name [String] the service name of this openstack instance
+    #   to lookup in configuration
+    def default_opts(service_name)
+      return  conf[:services, service_name.to_sym]
+    end
+
+    def to_seconds(string)
+      regex_m = /^(\d+)\s*(m|min|minutes|mins)+$/
+      regex_h = /^(\d+)\s*(h|hour|hours|hrs)+$/
+      regex_d = /^(\d+)\s*(d|day|days)+$/
+      regex_w = /^(\d+)\s*(w|week|weeks|wks)+$/
+      case string
+      when regex_m
+        return string.match(regex_m)[1].to_i * 60
+      when regex_h
+        return string.match(regex_h)[1].to_i * 60 * 60
+      when regex_d
+        return string.match(regex_d)[1].to_i * 24 * 60 * 60
+      when regex_w
+        return string.match(regex_w)[1].to_i * 7 * 24 * 60 * 60
+      else
+        raise "Cannot convert '#{string}' to seconds!"
+      end
+    end
+
+    # create a json which specifies OSD cluster
+    # in the future we plan to move the logic into the script 'osd-provision.sh'
+    def generate_json(name)
+      json_data = {
+        "name" => name,
+        "managed" => true,
+        "multi_az" => false,
+        "byoc" => false
+      }
+
+      if @multi_az
+        json_data.merge!({"multi_az" => @multi_az})
+      end
+
+      if @region
+        json_data.merge!({"region" => {"id" => @region}})
+      end
+
+      if @version
+        json_data.merge!({"version" => {"id" => "openshift-v#{@version}"}})
+      end
+
+      if @nodes
+        json_data.merge!({"nodes" => {"compute" => @nodes.to_i}})
+      end
+
+      if @lifespan
+        expiration = Time.now + to_seconds(@lifespan)
+        json_data.merge!({"expiration_timestamp" => expiration.strftime("%Y-%m-%dT%H:%M:%SZ")})
+      end
+
+      if @aws_account_id && @aws_access_key && @aws_secret_key
+        json_data.merge!({"aws" => {"account_id":@aws_account_id, "access_key_id":@aws_access_key, "secret_access_key":@aws_secret_key}})
+        json_data.merge!({"byoc" => true})
+      end
+
+      return json_data.to_json
+    end
+
+    # download the script 'osd-provision.sh' which takes care of the OSD installation/uninstallation
+    def download_osd_script
+      osd_repo_uri = ENV['GIT_OSD_URI'] || 'https://gitlab.cee.redhat.com/mk-bin-packing/mk-performance-tests.git'
```

Comment:
If this class relies on non-public resources, it is not really useful in a public open source project. If this class is needed for internal use cases, then it will need to be stored somewhere internally. Lets discuss this option in chat.

Reply:
ok, I did not realized this is a public repo :(

Reply:
So what do we do? We move this class to templates repo? Or any other ideas?

Reply:
I have refactored it so that it now directly uses https://github.com/openshift-online/ocm-cli